src/Strategy/CommonClassesStrategy.php

```diff
@@ -75,14 +75,14 @@ final class CommonClassesStrategy implements DiscoveryStrategy
             ['class' => SlimUriFactory::class, 'condition' => [SlimRequest::class, SlimUriFactory::class]],
         ],
         HttpAsyncClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Promise::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
+            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, HttpClient::class, Promise::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
             ['class' => Guzzle7::class, 'condition' => Guzzle7::class],
             ['class' => Guzzle6::class, 'condition' => Guzzle6::class],
             ['class' => Curl::class, 'condition' => Curl::class],
             ['class' => React::class, 'condition' => React::class],
         ],
         HttpClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
+            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, HttpClient::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
```

Comment:
Unfortunately this does not seem to change anything. [Symfony throws a `\LogicException`](https://github.com/symfony/http-client/blob/290eb481973b4984eb59377585f6afbc65a0f645/HttplugClient.php#L43-L49) outside of the class structure, so as soon as the file is loaded, and `php-http/httplug` is missing, the `\LogicException` is thrown.

Then in `\Http\Discovery\Strategy\CommonClassesStrategy::getPsr18Candidates`, that `\LogicException` is caught, and converted to a `trigger_error`:

https://github.com/php-http/discovery/blob/master/src/Strategy/CommonClassesStrategy.php#L139-L145

In trying to debug this, the only ways I found this was working was to turn that try/catch into silence (not ideal...):

```
            try {
                if (is_subclass_of($c['class'], Psr18Client::class)) {
                    $candidates[] = $c;
                }
            } catch (\Throwable $e) {
            }
```

but ALSO you would need to add `HttpClient::class` as a condition to `symfonyPsr18Instantiate` under the `Psr18Client::class` definitions:

```
        Psr18Client::class => [
            [
                'class' => [self::class, 'symfonyPsr18Instantiate'],
                'condition' => [HttpClient::class, SymfonyPsr18::class, Psr17RequestFactory::class],
            ],
```

Silencing the exception feels very wrong, but with a clear comment explaining why, it might be OK, as it doesn't look like there is a logger around to log a warning to.

Reply:
right the problem occurs before conditions are even checked. i realized that we need to use our safe class exists here as well. changed to use that and in my small reproducer that makes the trigger_error go away.